fdbclient/BlobCipher.cpp

```diff
@@ -940,14 +945,15 @@ Reference<EncryptBuf> EncryptBlobCipherAes265Ctr::encrypt(const uint8_t* plainte
 	ASSERT(isEncryptHeaderAuthTokenDetailsValid(authTokenMode, authTokenAlgo));
 
 	// Populate cipherText encryption-key details
-	header->cipherTextDetails.baseCipherId = textCipherKey->getBaseCipherId();
-	header->cipherTextDetails.encryptDomainId = textCipherKey->getDomainId();
-	header->cipherTextDetails.salt = textCipherKey->getSalt();
+	header->cipherTextDetails = textCipherKey->details();
 	// Populate header encryption-key details
-	// TODO: HeaderCipherKey is not necessary if AuthTokenMode == NONE
-	header->cipherHeaderDetails.encryptDomainId = headerCipherKey->getDomainId();
-	header->cipherHeaderDetails.baseCipherId = headerCipherKey->getBaseCipherId();
-	header->cipherHeaderDetails.salt = headerCipherKey->getSalt();
+	if (authTokenMode != ENCRYPT_HEADER_AUTH_TOKEN_MODE_NONE) {
+		header->cipherHeaderDetails = headerCipherKey->details();
+	} else {
+		header->cipherHeaderDetails.encryptDomainId = INVALID_ENCRYPT_DOMAIN_ID;
```

Comment:
nit: `header->cipherHeaderDetails` default construction does the same. We can remove the explicit assignment. 

Free free to do it as a separate PR (if is Clean and ready to merge)